shell/platform/darwin/ios/framework/Source/FlutterPlatformViewsTest.mm

```diff
@@ -266,16 +277,252 @@ - (void)testChildClippingViewShouldBeTheBoundingRectOfPlatformView {
-                kFloatCompareEpsilon);
-  XCTAssertTrue(fabs(platformViewRectInFlutterView.origin.y - childClippingView.frame.origin.y) <
-                kFloatCompareEpsilon);
-  XCTAssertTrue(fabs(platformViewRectInFlutterView.size.width -
-                     childClippingView.frame.size.width) < kFloatCompareEpsilon);
-  XCTAssertTrue(fabs(platformViewRectInFlutterView.size.height -
-                     childClippingView.frame.size.height) < kFloatCompareEpsilon);
+  XCTAssertLessThan(fabs(platformViewRectInFlutterView.origin.x - childClippingView.frame.origin.x),
+                    kFloatCompareEpsilon);
+  XCTAssertLessThan(fabs(platformViewRectInFlutterView.origin.y - childClippingView.frame.origin.y),
+                    kFloatCompareEpsilon);
+  XCTAssertLessThan(
+      fabs(platformViewRectInFlutterView.size.width - childClippingView.frame.size.width),
+      kFloatCompareEpsilon);
+  XCTAssertLessThan(
+      fabs(platformViewRectInFlutterView.size.height - childClippingView.frame.size.height),
+      kFloatCompareEpsilon);
+
+  flutterPlatformViewsController->Reset();
+}
+
+- (void)testClipRect {
+  flutter::FlutterPlatformViewsTestMockPlatformViewDelegate mock_delegate;
+  auto thread_task_runner = CreateNewThread("FlutterPlatformViewsTest");
+  flutter::TaskRunners runners(/*label=*/self.name.UTF8String,
+                               /*platform=*/thread_task_runner,
+                               /*raster=*/thread_task_runner,
+                               /*ui=*/thread_task_runner,
+                               /*io=*/thread_task_runner);
+  auto platform_view = std::make_unique<flutter::PlatformViewIOS>(
+      /*delegate=*/mock_delegate,
+      /*rendering_api=*/flutter::IOSRenderingAPI::kSoftware,
+      /*task_runners=*/runners);
+
+  auto flutterPlatformViewsController = std::make_unique<flutter::FlutterPlatformViewsController>();
+
+  FlutterPlatformViewsTestMockFlutterPlatformFactory* factory =
+      [[FlutterPlatformViewsTestMockFlutterPlatformFactory new] autorelease];
+  flutterPlatformViewsController->RegisterViewFactory(
+      factory, @"MockFlutterPlatformView",
+      FlutterPlatformViewGestureRecognizersBlockingPolicyEager);
+  FlutterResult result = ^(id result) {
+  };
+  flutterPlatformViewsController->OnMethodCall(
+      [FlutterMethodCall
+          methodCallWithMethodName:@"create"
+                         arguments:@{@"id" : @2, @"viewType" : @"MockFlutterPlatformView"}],
+      result);
+
+  XCTAssertNotNil(gMockPlatformView);
+
+  UIView* mockFlutterView = [[[UIView alloc] initWithFrame:CGRectMake(0, 0, 10, 10)] autorelease];
+  flutterPlatformViewsController->SetFlutterView(mockFlutterView);
+  // Create embedded view params
+  flutter::MutatorsStack stack;
+  // Layer tree always pushes a screen scale factor to the stack
+  SkMatrix screenScaleMatrix =
+      SkMatrix::MakeScale([UIScreen mainScreen].scale, [UIScreen mainScreen].scale);
+  stack.PushTransform(screenScaleMatrix);
```

Comment:
nit: does it need this to test clip specifically?

Reply:
yes. as the clip paths will need to get scaled as well.